internal/storage/image.go

```diff
@@ -398,6 +412,145 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+// nolint:staticcheck
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, configImage, userSpecifiedImage string, auth *runtimeapi.AuthConfig) error {
+	if userSpecifiedImage == "" {
+		if systemContext.SignaturePolicyPath == "" {
+			log.Infof(ctx, "User specified image is empty and no policy applied, skipping check if image is allowed to run")
+			return nil
+		}
+
+		return errors.New("user specified image is empty")
```

Comment:
Eventually this should probably be more helpful to users — ”Kubelet did not set the CRI Config.Image.UserSpecifiedImage field, please update it to use signature policies” or something.